mkosi/kmod.py

```diff
@@ -180,6 +180,11 @@ def gen_required_kernel_modules(
             mods = set(modulesd.rglob("*.ko*"))
         firmware = set()
 
+    links = {
```

Comment:
This is a bad variable name, these aren't links, they're resolved links.

I don't think we need a new variable name for this, just adding the resolved symlinks to the `firmware` variable should be good enough.

This can also use a general comment explaining why we're doing this.